File: CompanyApi/Models/Company.cs
using System.ComponentModel.DataAnnotations;

namespace CompanyApi.Models
{
    public class Company
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Exchange { get; set; }

        [Required]
        public string Ticker { get; set; }

        [Required]
        [MinLength(12)]
        [MaxLength(12)]
        [RegularExpression(@"^[A-Z]{2}[A-Z0-9]{10}$", ErrorMessage = "ISIN must be 12 characters, start with two uppercase letters, and contain only uppercase letters or numbers.")]
        public string Isin { get; set; }

        public string? Website { get; set; }
    }
}

File: company-frontend/src/app/domains/companies/models/company.ts
import { Company as CompanyApiModel } from '@api/models/company';

export interface Company {
    exchange: string;
    id: number;
    isin: string;
    name: string;
    ticker: string;
    website?: string | null;
}

/**
 * Maps an API Company model to a UI Company model.
 * @param apiCompany The API Company model to map.
 * @returns The mapped UI Company model.
 */
export function mapApiCompanyToUi(apiCompany: CompanyApiModel): Company {
  return {
    exchange: apiCompany.exchange,
    id: apiCompany.id,
    name: apiCompany.name,
    isin: apiCompany.isin,
    ticker: apiCompany.ticker,
    website: apiCompany.website ?? null
  };
}